src/config.rs

```diff
@@ -291,11 +303,49 @@ pub struct DirectoriesConfig {
     pub custom_pages_dir: Option<PathWithSource>,
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum RawTlsBackend {
+    /// Native TLS (`SChannel` on Windows, Secure Transport on macOS and OpenSSL otherwise)
+    NativeTls,
+    /// Rustls with `WebPKI` roots.
+    RustlsWithWebpkiRoots,
+    /// Rustls with native roots.
+    RustlsWithNativeRoots,
+}
+
+impl Default for RawTlsBackend {
+    fn default() -> Self {
+        *SUPPORTED_TLS_BACKENDS.first().unwrap()
+    }
+}
+
+impl std::fmt::Display for RawTlsBackend {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        self.serialize(f)
+    }
+}
+
+/// Allows choosing a `reqwest`'s TLS backend. Available TLS backends:
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum TlsBackend {
+    /// Native TLS (`SChannel` on Windows, Secure Transport on macOS and OpenSSL otherwise)
+    #[cfg(feature = "native-tls")]
+    NativeTls,
+    /// Rustls with `WebPKI` roots.
+    #[cfg(feature = "rustls-with-webpki-roots")]
+    RustlsWithWebpkiRoots,
+    /// Rustls with native roots.
+    #[cfg(feature = "rustls-with-native-roots")]
+    RustlsWithNativeRoots,
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Config {
     pub style: StyleConfig,
     pub display: DisplayConfig,
     pub updates: UpdatesConfig,
+    pub tls_backend: TlsBackend,
```

Comment:
I think we can move `tls_backend` into the `UpdatesConfig` instead of having it directly in `Config`